pkgs/by-name/ju/just/package.nix

```diff
@@ -87,16 +83,31 @@ rustPlatform.buildRustPackage rec {
     "--skip=shebang::run_shebang" # test case very rarely fails with "Text file busy"
   ];
 
-  postInstall = ''
-    mkdir -p $doc/share/doc/$name
-    mv ./book/en/build/html $doc/share/doc/$name
-    installManPage man/just.1
-
-    installShellCompletion --cmd just \
-      --bash completions/just.bash \
-      --fish completions/just.fish \
-      --zsh completions/just.zsh
-  '';
+  postInstall =
+    let
+      just = "${stdenv.hostPlatform.emulator buildPackages} $out/bin/just";
+    in
+    lib.optionalString withDocs (
+      if canExecute then
+        ''
+          mkdir -p $doc/share/doc/$name
+          mv ./book/en/build/html $doc/share/doc/$name
+        ''
+      else
+        ''
+          mkdir -p $doc/share
+          cp -r ${buildPackages.just.doc}/share/doc $doc/share
+        ''
+    )
+    + lib.optionalString (stdenv.hostPlatform.emulatorAvailable buildPackages) ''
+      ${just} --man > ./just.1
+      installManPage ./just.1
```

Comment:
Why aren't we copying this from buildPackages.just as well?

Reply:
I've looked back on the solution with the copying and it doesn't really sit well with me. 
I refactored `generate-book` to use emulation. As it turns out it's not a bottleneck at all and takes no time at all.
Sorry for the back and forth

Reply:
Emulation should really only be used where it's a last resort, because we don't have emulators for every platform we might want to cross compile to.

Reply:
We've discussed it a bit here https://github.com/NixOS/nixpkgs/pull/351032, but I don't have a clear opinion on this

Reply:
Wrong link?

Reply:
Oh https://github.com/NixOS/nixpkgs/issues/308283